Adafruit_MQTT.cpp

```diff
@@ -116,7 +116,7 @@ Adafruit_MQTT::Adafruit_MQTT(const char *server, uint16_t port, const char *cid,
   will_qos = 0;
   will_retain = 0;
 
-  packet_id_counter = 0;
+  packet_id_counter = 1; // MQTT spec forbids packet id of 0 if QOS=1
```

Comment:
@dlizotte-uwo this constructor does not guarantee a QoS level, we should handle checking and incrementing the PID based on the QoS level within the SUBSCRIBE, UNSUBSCRIBE, and PUBLISH calls.

> SUBSCRIBE, UNSUBSCRIBE, and PUBLISH (in cases where QoS > 0) Control Packets MUST contain a non-zero 16-bit Packet Identifier [MQTT-2.3.1-1].

Reply:
@brentru Hi, the QoS only needs to be checked for PUBLISH (and is already done in [L707](https://github.com/adafruit/Adafruit_MQTT_Library/blob/7b2061ad85e9ba322048fb53a5e30324bd88334f/Adafruit_MQTT.cpp#L707)). 
SUBSCRIBE and UNSUBSCRIBE always need a (valid) packet id
[(Ref)](https://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc385349217)
In PUBLISH QoS = 0, the packet id is not used.